### PR TITLE
store: use simple LRU cache, add metrics, evict on size

### PR DESCRIPTION
**Pull request by fabxc**
@Bplotka

### CODE REVIEW

File: pkg/store/cache.go [hunk 1]
```diff
@@ -1,47 +1,159 @@
...
+	l, err := lru.NewLRU(1e12, func(key, val interface{}) {
+		k := key.(cacheItem).keyType()
+		v := val.([]byte)
+
+		evicted.WithLabelValues(k).Inc()
+		c.current.WithLabelValues(k).Dec()
+		c.currentSize.WithLabelValues(k).Sub(float64(len(v)))
+
+		c.curSize -= uint64(len(v))
+	})
```
Comment (fabxc) [inline pkg/store/cache.go]: This is the eviction callback of the LRU implementation.

File: pkg/store/cache.go [hunk 2]
```diff
@@ -1,47 +1,159 @@
...
+	cacheTypeSeries   = "series"
+)
+
 type cacheItem struct {
 	block ulid.ULID
 	key   interface{}
 }
+
+func (c cacheItem) keyType() string {
+	switch c.key.(type) {
```
Comment (Bplotka) [inline pkg/store/cache.go]: How does it work?

https://play.golang.org/p/tHmYrn-SsR

Comment (fabxc) [reply pkg/store/cache.go]: Your playground code checks the `cacheType*` string. Here we check `cacheKey*`.

Comment (Bplotka) [reply pkg/store/cache.go]: ah did see the `Key` word there, ok

File: pkg/store/cache.go [hunk 3]
```diff
@@ -1,47 +1,162 @@
...
+
+		c.curSize -= uint64(len(v))
+	})
 	if err != nil {
 		return nil, err
 	}
-	return &indexCache{lru: l}, nil
+	c.lru = l
+
+	fmt.Println("reg", reg)
```
Comment (Bplotka) [inline pkg/store/cache.go]: Debug prints?

File: pkg/store/cache.go (hunk 1 above)
Comment (Bplotka) [reply pkg/store/cache.go]: Worth to add a comment or construct the function in separate line with the name then (:

**Comment by fabxc** [comment]
Done

**Comment by fabxc** [comment]
One is green :P Other the known flake.

**Pull request merged**